public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -17,18 +17,18 @@ const SubmissionsTable = (): JSX.Element => {
             assignments.push(...state.assignments[courseID])
         }
         assignments.sort((a, b) => {
-            if (b.deadline > a.deadline) {
+            if ((a.deadline && b.deadline) && (b.deadline > a.deadline)) {
```

Comment:
I was going to suggest this as more readable code:
```ts
            if (!a.deadline || !b.deadline) {
                return 0
            }
            if (a.deadline < b.deadline) {
                return -1
            }
            if (a.deadline > b.deadline) {
                return 1
            }
```

But then copilot suggested this:
```typescript
            if (a.deadline && b.deadline) {
                return a.deadline.toDate().getTime() - b.deadline.toDate().getTime()
            }
            return 0
```
I also tried:
```ts
       return a.deadline.seconds - b.deadline.seconds
```
But `seconds` are `bigint` (not a number) and didn't work with the sort api.

Reply:
Seems like there is prior art for doing it as copilot suggested. See `Helpers.ts`:
```ts
        case EnrollmentSort.Activity:
            if (a.lastActivityDate && b.lastActivityDate) {
                return sortOrder * (a.lastActivityDate.toDate().getTime() - b.lastActivityDate.toDate().getTime()) 
            }
            return 0               
```